### PR TITLE
Update getResolverStrategy for rootRequirements

### PR DESCRIPTION
**Pull request by BlythMeister**
When root requirement, us the resolver strategy for direct dependencies.
Even if it also has other requirements

Attempted fix for #3669 

And meaning that the statement `that all direct dependencies will still get their latest matching versions, no matter the value of the strategy option. If you want to influence the resolution of direct dependencies then read about the lowest_matching option.` in the docs is actually correct

### CODE REVIEW

**Comment by BlythMeister** [comment]
@forki this change does update tests...but i think that the tests contradict the documented behaviour.

In these tests, the dependencies are specified in the paket.dependencies file
However, we do not expect to get the latest version

**Comment by BlythMeister** [comment]
Is there anything more needed from me to get this merged?
Looks like the build went green 🙂

File: tests/Paket.Tests/Resolver/GlobalOptimisticStrategySpecs.fs [hunk 1]
```diff
@@ -159,7 +159,7 @@ let ``should favor strategy from parent that overrides strategy``() =
     let resolved =
         DependenciesFile.FromSource(config)
         |> resolve graph2 UpdateMode.UpdateAll
-    getVersion resolved.[PackageName "Castle.Windsor"] |> shouldEqual "3.2.1"
+    getVersion resolved.[PackageName "Castle.Windsor"] |> shouldEqual "3.3.0"
```
Comment (forki) [inline tests/Paket.Tests/Resolver/GlobalOptimisticStrategySpecs.fs]: why is this increased? I thought we use min strategy?

Comment (BlythMeister) [reply tests/Paket.Tests/Resolver/GlobalOptimisticStrategySpecs.fs]: this test:
```
[<Test>]
let ``should favor strategy from parent that overrides strategy``() = 
    let config = """
    strategy max
    source http://www.nuget.org/api/v2
    nuget Nancy.Bootstrappers.Windsor !~> 0.23
    nuget Castle.Windsor
    nuget Castle.Windsor-NLog !> 0
    """
    let resolved =
        DependenciesFile.FromSource(config)
        |> resolve graph2 UpdateMode.UpdateAll
    getVersion resolved.[PackageName "Castle.Windsor"] |> shouldEqual "3.3.0"
    getVersion resolved.[PackageName "Castle.Windsor-NLog"] |> shouldEqual "3.3.0"
    getVersion resolved.[PackageName "Castle.Core-NLog"] |> shouldEqual "3.3.0"
    getVersion resolved.[PackageName "Castle.Core"] |> shouldEqual "3.3.0"
    getVersion resolved.[PackageName "Nancy.Bootstrappers.Windsor"] |> shouldEqual "0.23"
```
is set to max but more importantly has Castle.Windsor as a top level dependency.  So should take the latest version

**Comment by forki** [comment]
thank you

**Pull request merged**